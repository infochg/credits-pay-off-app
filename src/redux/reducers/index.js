import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';

import notifications from './notifications';
import loading from './loading';
import registrationError from './registrationError';
import token from './token';
import userInfo from './userInfo';
import overviewData from './overviewData';
import paymentsData from './paymentsData';
import transactionsData from './transactionsData';
import subscriptionsData from './subscriptionsData';
import notificationsData from './notificationsData';
import accountsData from './accountsData';
import isContactSent from './isContactSent';
import isResetCodeVerified from './isResetCodeVerified';
import isPasswordReset from './isPasswordReset';
import isPasswordChanged from './isPasswordChanged';
import isAddAccountOpen from './isAddAccountOpen';
import walkthrough from './walkthrough';
import is2FA from './is2FA';
import isWrongImageReqSent from './isWrongImageReqSent';
import isUserSubscribed from './isUserSubscribed';
import isUserUnsubscribed from './isUserUnsubscribed';
import billingDate from './billingDate';

const rootReducer = combineReducers({
  loading,
  notifications,
  registrationError,
  token,
  userInfo,
  overviewData,
  paymentsData,
  transactionsData,
  subscriptionsData,
  notificationsData,
  accountsData,
  isContactSent,
  isResetCodeVerified,
  isPasswordReset,
  isPasswordChanged,
  isAddAccountOpen,
  walkthrough,
  is2FA,
  isWrongImageReqSent,
  isUserSubscribed,
  isUserUnsubscribed,
  billingDate,
  form: formReducer
});

export default (state, action) =>
  rootReducer(action.type === 'RESET' ? undefined : state, action);
