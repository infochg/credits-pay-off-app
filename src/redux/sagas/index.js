import { takeEvery } from 'redux-saga/effects';
import {
  LOGOUT_REQUEST,
  REGISTRATION_REQUEST,
  SIGNIN_REQUEST,
  VERIFY_PHONE_REQUEST,
  USER_INFO_REQUEST,
  OVERVIEW_DATA_REQUEST,
  PAYMENTS_DATA_REQUEST,
  TRANSACTIONS_DATA_REQUEST,
  SUBSCRIPTIONS_DATA_REQUEST,
  ACCOUNTS_DATA_REQUEST,
  NOTIFICATIONS_DATA_REQUEST,
  NOTIFICATIONS_CLEAR_REQUEST,
  CHANGE_PASSWORD_REQUEST,
  FORGOT_PASSWORD_REQUEST,
  VERIFY_RESET_CODE_REQUEST,
  RESET_PASSWORD_REQUEST,
  UPDATE_USER_INFO_REQUEST,
  SEND_LOG_REQUEST,
  SEND_2FA_REQUEST,
  TOGGLE_2FA_REQUEST,
  WRONG_IMAGE_REQUEST,
  ADD_EMPATHIZE_SUBSCRIPTION_REQUEST,
  CANCEL_EMPATHIZE_SUBSCRIPTION_REQUEST,
  RESET_EMPATHIZE_SUBSCRIPTION_REQUEST,
  BILLING_DATE_REQUEST
} from '../actions/actionTypes';

import sagaLogout from './sagaLogout';
import sagaRegistration from './sagaRegistration';
import sagaVerifyPhone from './sagaVerifyPhone';
import sagaSignin from './sagaSignin';
import sagaUserInfo from './sagaUserInfo';
import sagaUpdateUserInfo from './sagaUpdateUserInfo';
import sagaOverviewData from './sagaOverviewData';
import sagaPaymentsData from './sagaPaymentsData';
import sagaTransactionsData from './sagaTransactionsData';
import sagaSubscriptionsData from './sagaSubscriptionsData';
import sagaAccountsData from './sagaAccountsData';
import sagaNotificationsData from './sagaNotificationsData';
import sagaClearNotifications from './sagaClearNotifications';
import sagaChangePassword from './sagaChangePassword';
import sagaForgotPassword from './sagaForgotPassword';
import sagaVerifyResetCode from './sagaVerifyResetCode';
import sagaResetPassword from './sagaResetPassword';
import sagaSendLog from './sagaSendLog';
import sagaSend2FACode from './sagaSend2FACode';
import sagaToggle2FA from './sagaToggle2FA';
import sagaWrongImage from './sagaWrongImage';
import sagaAddEmpathizeSubscription from './sagaAddEmpathizeSubscription';
import sagaCancelEmpathizeSubscription from './sagaCancelEmpathizeSubscription';
import sagaResetEmpathizeSubscription from './sagaResetEmpathizeSubscription';
import sagaBillingDate from './sagaBillingDate';

export default function* sagas() {
  yield takeEvery(LOGOUT_REQUEST, sagaLogout);
  yield takeEvery(REGISTRATION_REQUEST, sagaRegistration);
  yield takeEvery(VERIFY_PHONE_REQUEST, sagaVerifyPhone);
  yield takeEvery(SIGNIN_REQUEST, sagaSignin);
  yield takeEvery(USER_INFO_REQUEST, sagaUserInfo);
  yield takeEvery(UPDATE_USER_INFO_REQUEST, sagaUpdateUserInfo);
  yield takeEvery(OVERVIEW_DATA_REQUEST, sagaOverviewData);
  yield takeEvery(PAYMENTS_DATA_REQUEST, sagaPaymentsData);
  yield takeEvery(TRANSACTIONS_DATA_REQUEST, sagaTransactionsData);
  yield takeEvery(SUBSCRIPTIONS_DATA_REQUEST, sagaSubscriptionsData);
  yield takeEvery(ACCOUNTS_DATA_REQUEST, sagaAccountsData);
  yield takeEvery(NOTIFICATIONS_DATA_REQUEST, sagaNotificationsData);
  yield takeEvery(NOTIFICATIONS_CLEAR_REQUEST, sagaClearNotifications);
  yield takeEvery(CHANGE_PASSWORD_REQUEST, sagaChangePassword);
  yield takeEvery(FORGOT_PASSWORD_REQUEST, sagaForgotPassword);
  yield takeEvery(VERIFY_RESET_CODE_REQUEST, sagaVerifyResetCode);
  yield takeEvery(RESET_PASSWORD_REQUEST, sagaResetPassword);
  yield takeEvery(SEND_LOG_REQUEST, sagaSendLog);
  yield takeEvery(SEND_2FA_REQUEST, sagaSend2FACode);
  yield takeEvery(TOGGLE_2FA_REQUEST, sagaToggle2FA);
  yield takeEvery(WRONG_IMAGE_REQUEST, sagaWrongImage);
  yield takeEvery(
    ADD_EMPATHIZE_SUBSCRIPTION_REQUEST,
    sagaAddEmpathizeSubscription
  );
  yield takeEvery(
    CANCEL_EMPATHIZE_SUBSCRIPTION_REQUEST,
    sagaCancelEmpathizeSubscription
  );
  yield takeEvery(
    RESET_EMPATHIZE_SUBSCRIPTION_REQUEST,
    sagaResetEmpathizeSubscription
  );
  yield takeEvery(BILLING_DATE_REQUEST, sagaBillingDate);
}
