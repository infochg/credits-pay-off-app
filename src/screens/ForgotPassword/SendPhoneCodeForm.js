import React from 'react';
import { Field, reduxForm, Form } from 'redux-form';
import { Button, CircularProgress, Grid } from '@material-ui/core';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '../../components/Common/TextField';
import ErrorBoundary from '../../containers/ErrorBoundary';

const required = value => (value && value !== '' ? undefined : 'Required');

const useStyles = makeStyles(theme => ({
  submitBtn: {
    marginTop: '40px'
  },
  forgotText: {
    textAlign: 'center',
    '& a': {
      textDecoration: 'none',
      color: theme.palette.primary.main,
      '&:hover': {
        textDecoration: 'underline'
      }
    }
  }
}));

function SendPhoneCodeForm(props) {
  const classes = useStyles();
  const { handleSubmit, pristine, submitting, loading } = props;

  try {
    return (
      <Form onSubmit={handleSubmit}>
        <Grid container>
          <Grid item xs={12}>
            <Field
              name="code"
              component={TextField}
              placeholder="Email or Phone number"
              validate={required}
              variant="outlined"
              fullWidth
            />
          </Grid>
        </Grid>
        {!loading ? (
          <Button
            variant="contained"
            color="primary"
            disableElevation
            type="submit"
            disabled={pristine || submitting}
            className={classes.submitBtn}
          >
            Send Code
          </Button>
        ) : (
          <div className={classes.submitBtn}>
            <CircularProgress />
          </div>
        )}
      </Form>
    );
  } catch (e) {
    return <ErrorBoundary error={e} />;
  }
}

SendPhoneCodeForm.propTypes = {
  handleSubmit: PropTypes.func,
  pristine: PropTypes.bool,
  submitting: PropTypes.bool,
  classes: PropTypes.shape({}),
  loading: PropTypes.bool
};

SendPhoneCodeForm.defaultProps = {
  handleSubmit: undefined,
  pristine: undefined,
  submitting: undefined,
  classes: undefined,
  loading: undefined
};

export default reduxForm({
  form: 'SendPhoneCodeForm'
})(SendPhoneCodeForm);
