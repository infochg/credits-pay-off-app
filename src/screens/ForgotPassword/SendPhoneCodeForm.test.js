import React from 'react';
import { createStore, combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import SendPhoneCodeForm from './SendPhoneCodeForm';

describe('<SendPhoneCodeForm />', () => {
  const fieldReducer = (state = {}, { type }) => {
    switch (type) {
      default:
        return state;
    }
  };

  const reducer = combineReducers({
    field: fieldReducer,
    form: formReducer
  });

  let wrapper;

  beforeEach(() => {
    const store = createStore(reducer);

    wrapper = mount(
      <Provider store={store}>
        <SendPhoneCodeForm />
      </Provider>
    );
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should have a сщву field', () => {
    expect(
      wrapper.find('input[placeholder="Email or Phone number"]').length
    ).toEqual(1);
  });

  it('fills the input with a default value (empty)', () => {
    expect(
      wrapper.find('input[placeholder="Email or Phone number"]').prop('value')
    ).toBe('');
  });

  it('updates input value when changed', () => {
    const event = { target: { value: 'Test' } };
    wrapper
      .find('input[placeholder="Email or Phone number"]')
      .simulate('change', event);
    expect(
      wrapper.find('input[placeholder="Email or Phone number"]').prop('value')
    ).toBe('Test');
  });
});
