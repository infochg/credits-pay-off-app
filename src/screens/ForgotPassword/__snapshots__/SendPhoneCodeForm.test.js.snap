// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SendPhoneCodeForm /> should match the snapshot 1`] = `"<form><div class=\\"MuiGrid-root MuiGrid-container\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"makeStyles-wrapper-106\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root makeStyles-label-108 MuiInputLabel-animated\\"></label><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" name=\\"code\\" placeholder=\\"Email or Phone number\\" type=\\"text\\" class=\\"MuiInputBase-input MuiOutlinedInput-input makeStyles-input-107\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"PrivateNotchedOutline-root-167 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legend-168\\" style=\\"width: 0.01px;\\"><span>â€‹</span></legend></fieldset></div></div></div></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submitBtn-1 MuiButton-containedPrimary MuiButton-disableElevation Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"submit\\" disabled=\\"\\"><span class=\\"MuiButton-label\\">Send Code</span></button></form>"`;
